plugins/pulp_rpm/plugins/importers/yum/parse/rpm.py

```diff
@@ -60,19 +59,20 @@ def change_location_tag(primary_xml_snippet, relpath):
     Transform the <location> tag to strip out leading directories and add `Packages/<first_letter>`.
 
     :param primary_xml_snippet: snippet of primary xml text for a single package
-    :type  primary_xml_snippet: unicode
+    :type  primary_xml_snippet: str
 
     :param relpath: Package's 'relativepath'
-    :type  relpath: unicode
+    :type  relpath: str
     """
-
+    primary_xml_snippet = primary_xml_snippet.decode('utf-8', 'replace')
```

Comment:
I put decoding back inside the `change_location_tag` because this function is used in multiple places.
The other option is to decode xml snippet each time before calling `change_location_tag`.